DFC.App.Triagetool.Services.CacheContentService/CacheReloadService.cs

```diff
@@ -0,0 +1,149 @@
+
+                    var documents = GetDocumentOptionModel(options, pages);
+
+                    await UpdateDocuments(documents).ConfigureAwait(false);
+                }
+
+                logger.LogInformation("Reload cache completed");
+            }
+            catch (Exception ex)
+            {
+                logger.LogError(ex, "Unable to reload Triage Tool content from the CMS");
+            }
+        }
+
+        private async Task<IList<TriageToolOptionItemModel>> GetTriageToolOptions()
+        {
+            var triageToolOptions = new List<TriageToolOptionItemModel>();
+            var optionSummary = await cmsApiService.GetSummaryAsync<TriageToolOptionSummaryModel>(CmsContentKeyHelper.OptionTag)
+                .ConfigureAwait(false);
+
+            if (optionSummary != null && optionSummary.Any())
+            {
+                foreach (var option in optionSummary)
+                {
+                    var triageToolOption = await cmsApiService.GetItemAsync<TriageToolOptionItemModel>(option.Url!)
+                        .ConfigureAwait(false);
+                    if (triageToolOption != null)
+                    {
+                        triageToolOptions.Add(triageToolOption);
+                    }
+                }
+            }
+
+            return triageToolOptions;
+        }
+
+        private async Task<IList<CmsApiDataModel>> GetTriagePages()
+        {
+            var pages = new List<CmsApiDataModel>();
+
+            var pagesSummary = await cmsApiService.GetSummaryAsync<CmsApiSummaryItemModel>(CmsContentKeyHelper.PageTag)
+                .ConfigureAwait(false);
+
+            if (pagesSummary != null && pagesSummary.Any())
+            {
+                foreach (var pageItem in pagesSummary)
+                {
+                    var page = await cmsApiService.GetItemAsync<CmsApiDataModel>(pageItem.Url!).ConfigureAwait(false);
+                    if (page != null && page.UseInTriageTool)
+                    {
+                        pages.Add(page);
+                    }
+                }
+            }
+
+            return pages;
+        }
+
+        private IList<TriageToolOptionDocumentModel> GetDocumentOptionModel(IList<TriageToolOptionItemModel> options, IList<CmsApiDataModel> pages)
+        {
+            var pageDocuments = mapper.Map<IList<PageDocumentModel>>(pages);
+            var optionsDocuments = mapper.Map<IList<TriageToolOptionDocumentModel>>(options);
+
+            foreach (var optionDocument in optionsDocuments)
+            {
+                var pagesForOption = pageDocuments.Where(pd => optionDocument.FilterIds.Any(od => pd.Filters.Contains(od))).ToList();
+
+                if (pagesForOption.Any())
+                {
+                    optionDocument.Pages = pagesForOption.ToList();
+                    optionDocument.PageIds = pagesForOption.Select(s => s.Uri?.ToString() ?? string.Empty).ToList();
```

Comment:
Rather than use string.Empty for null Uri items, I would rather filter them out with a here clause so that there are no blank items in the PageIds?
Might be worth filtering out the null uri items from the pagesForOption on line 113?